opensrp-app/src/main/java/org/smartregister/service/HTTPAgent.java

```diff
@@ -189,48 +195,59 @@ public void setupHttpClient() {
-            int statusCode = httpResponse.getStatusLine().getStatusCode();
+            URL url = new URL(requestURL);
+            urlConnection = (HttpURLConnection) url.openConnection();
+            if (urlConnection instanceof HttpsURLConnection) {
+                OpensrpSSLHelper opensrpSSLHelper = new OpensrpSSLHelper(context, configuration);
+                ((HttpsURLConnection) urlConnection).setSSLSocketFactory(opensrpSSLHelper.getSSLSocketFactory());
+            }
+            urlConnection.setConnectTimeout(CONNECTION_TIMEOUT);
+            urlConnection.setReadTimeout(CONNECTION_TIMEOUT);
+            final String basicAuth = "Basic " + Base64.encodeToString((userName + ":" + password).getBytes(), Base64.NO_WRAP);
+            urlConnection.setRequestProperty("Authorization", basicAuth);
+            int statusCode = urlConnection.getResponseCode();
+            InputStream inputStream;
+            if (statusCode >= HttpStatus.SC_BAD_REQUEST)
+                inputStream = urlConnection.getErrorStream();
+            else
+                inputStream = urlConnection.getInputStream();
+            String responseString = IOUtils.toString(inputStream);
             if (statusCode == HttpStatus.SC_OK) {
-                LoginResponseData responseData = getResponseBody(httpResponse);
+                LoginResponseData responseData = getResponseBody(responseString);
                 loginResponse = retrieveResponse(responseData);
-            } else {
-                if (statusCode == HttpStatus.SC_UNAUTHORIZED) {
-                    logError("Invalid credentials for: " + userName + " using " + requestURL);
-                    loginResponse = UNAUTHORIZED;
-                } else {
-                    logError("Bad response from Dristhi. Status code:  " + statusCode + " username: "
-                            + userName + " using " + requestURL);
-                    loginResponse = UNKNOWN_RESPONSE;
-                }
-
-                String responseString = getResponseString(httpResponse);
+            } else if (statusCode == HttpStatus.SC_UNAUTHORIZED) {
+                logError("Invalid credentials for: " + userName + " using " + requestURL);
+                loginResponse = UNAUTHORIZED;
+            } else if (StringUtils.isNotBlank(responseString)) {
+                responseString = StringUtils.substringBetween(responseString, "<p><b>message</b>", "</u></p>");
                 if (StringUtils.isNotBlank(responseString)) {
-                    responseString = StringUtils.substringBetween(responseString, "<p><b>message</b>", "</u></p>");
-                    if (StringUtils.isNotBlank(responseString)) {
-                        responseString = responseString.replace("<u>", "").trim();
-                        loginResponse = CUSTOM_SERVER_RESPONSE.withMessage(responseString);
-                    }
+                    responseString = responseString.replace("<u>", "").trim();
```

Comment:
Maybe move the html string to a descriptive constant?

Reply:
I have added comments to describe what the code does